src/verifiable/certificate-chain-validator.ts

```diff
@@ -0,0 +1,113 @@
+import { promises } from 'fs'
+import * as tls from 'tls'
+
+import { JWS } from 'jose'
+import * as forge from 'node-forge'
+
+import { getJwkFromRecipient, isX5C } from './keys'
+
+import certificateFromPem = forge.pki.certificateFromPem
+
+/**
+ * Service to validate the certificate chain (using web PKI) for Verifiable PayIDs
+ * signed with a a server key.
+ */
+export default class CertificateChainValidator {
+  public caStore: forge.pki.CAStore
+
+  public constructor() {
+    this.caStore = forge.pki.createCaStore([])
+    this.importNodeRootCertficates()
+  }
+
+  /**
+   * Verifies the certificate chain for any x5c certificates inside the JWS protected section.
+   *
+   * @param jws - The JWS to verify.
+   * @returns True if verified.
+   */
+  public verifyCertificateChainJWS(jws: JWS.GeneralJWS): boolean {
+    return jws.signatures
+      .map((recipient) => this.verifyCertificateChainRecipient(recipient))
+      .every((val) => val)
```

Comment:
What is this `every` for here?

Reply:
it's just a way to check that all the values in the mapped array are true. For example `[true, true, true].every((val) => val)` will evaluate to `true` but `[true, false, true].every((val) => val)` will evaluate to `false`